REP-2006: ROS 2 Security Vulnerability Disclosure Policy. (#262)
ros-infrastructure/rep: rep-2006.rst

REP: 2006
Title: ROS 2 Vulnerability Disclosure Policy
Author: ROS 2 Security Working Group, Chris Lalancette <clalancette@openrobotics.org>
Status: Active
Type: Informational
Content-Type: text/x-rst
Created: 12-May-2020
Post-History:

Abstract
========

> This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.

Author: Yeah, the scope of this was one of my concerns as well.  We can't possibly be responsible for evaluating and fixing security bugs in the millions of lines of code covered by REP-2005.

That being said, the main intention here is to give us time to react before disclosure.  For most of the heavily-used packages, I would expect that the maintainers would be responsive and work to fix the issue.  If there are packages that have gone inactive, and the developers don't respond, there's not too much we can do about it.  Maybe at that point the response is to delist the package from REP-2005.

Your point about how we contact the maintainers and how they should respond is a good one.  I don't think it is reasonable to expect that every project setup a `security@` closed mailing list.  I think the best we can do in general is to directly contact the maintainers listed in the package.xml (though I'm open to other suggestions as well).  I'll add some verbiage around that to start with.

> This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.

Author: >  I'll add some verbiage around that to start with.

Actually, though, I don't think this is the right place for that.  This REP is a statement from the developers of ROS to the wider world, not a documentation of the internal processes.  If we are going to add verbiage, I would think it would be in REP-2005.  I'll add something there instead.

> This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.

Reviewer: > Actually, though, I don't think this is the right place for that. This REP is a statement from the developers of ROS to the wider world, not a documentation of the internal processes. 

I agree. There are two aspects to handling vuln reports: the process we go through with the reporter, and the internal process of actually triaging and fixing the vuln. This is the former, and I think it's important for it to stay that way. The latter still needs to be written, and I'm not sure it belongs in REP-2005 either. Maybe the developer guide?

> This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.

Reviewer 2: Yeah I agree it makes sense to have it elsewhere. Could be a process REP or something less formal if we think it requires iterations / maturing. This comment is mostly pointing out that this was a pending question when we drafted this document and that this REP does imply a process that is currently not defined anywhere. It would need to be for this REP to take effect

> This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.

Author: That kind of internal process documentation seem to me like it would go in a developer guide or a security group working document.  It doesn't exactly seem like an REP.  I would suggest a new page on https://github.com/ros2/ros2_documentation, but other places are fine as well.

Given that I have no familiarity with what these processes are expected to be, would someone be willing to step up and write down some of what has been discussed as the internal process for handling security reports?

> This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.

Reviewer 3: This and a handful of open issues raised in the discussions (e.g., how to handle embargoed issues, how deep in the dependency stack we travel, providing training materials, providing VDP tempates to vendors). I'll gather these from the [WG Wiki Page](http://wiki.ros.org/action/fullsearch/ROS2/WorkingGroups/Security) and add them to the working agenda.

> It would need to be for this REP to take effect

I'd like to move this forward as we continue to work through the supporting processes, similar to how a VDP was required with REP 2005 before this doc was issued.

> This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.

Author: Excellent, thanks for taking the lead here @SidFaber.  It's appreciated.

> This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.

Author: Given the above discussion, should I roll back the changes on REP-2005?

> This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.

Author: > I'd like to move this forward as we continue to work through the supporting processes, similar to how a VDP was required with REP 2005 before this doc was issued.

Yeah, that works fine for me.  I don't think having those processes in place is a prerequisite for merging this.

> This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.

Author: I'm going to resolve this conversation, as there is no further action to be taken on this PR for this.  We'll discuss the ongoing internal processes in the Security Working Group meeting.  Feel free to unresolve and/or add another comment if you feel differently.

Motivation
==========

> The developers of the Robot Operating System (ROS) take security very seriously.

Author: I did think about that.  The thing is that I think we are "developers of ROS", its just that the scope of what is covered in this document is ROS 2.  So I'd prefer to leave this as-is.

> The developers of the Robot Operating System (ROS) take security very seriously.

Reviewer 2: It's just such a common thing to hear in the community "ROS development didn't care about security" that it seems a low hanging fruit to disambiguate here for a potentially significant perception impact. We do know this REP only applies to ROS 2 but it can be overlooked and consider REPs as applying to the entire ROS ecosystem. For example REP 2004 has ROS 2 in mind but doesn't explicitly call out its scope to be ROS 2 and can thus be applied to ROS 1.

> The developers of the Robot Operating System (ROS) take security very seriously.

Author: The difference here is that we explicitly call out the scope to be REP-2005, so in my thinking there isn't really any ambiguity about what is covered.

> The developers of the Robot Operating System (ROS) take security very seriously.

Author: I know we don't agree on this, but I don't see a way out of the current stalemate since both sides are a matter of opinion.  I'm going to resolve this conversation to move forward, but feel free to reopen or keep commenting.

As such, we would like to be informed when a security bug is found so that it can be fixed and disclosed as quickly as possible.
The rest of the document outlines what is covered by this policy and how to report security vulnerabilities.

Scope
=====

The Vulnerability Disclosure Program outlined here covers all code within the ROS 2 Common Packages as defined in REP 2005.

> The ROS Security Working Group may also be able to assist you with reporting vulnerabilities which are not in scope of this policy but which are related to ROS or more generally to robotics.

Author: > Could this be done by the same entity / group of people as the people processing the reports?

By "done", do you mean helping to contact the relevant companies, or doing the engineering work?

> 
> If not, how would one contact the working group for these not-covered-by-this-rep vulnerabilities?

I believe the intention here is to say "if you find a vulnerability in a robot/software not directly related to this REP, we may know the people who are responsible and can direct you there".  I would think the reporting process would be the same as outlined below, its just that the ROS 2 security group wouldn't take any direct action other than redirecting.

But I didn't write this.  @SidFaber, any feedback here?

> The ROS Security Working Group may also be able to assist you with reporting vulnerabilities which are not in scope of this policy but which are related to ROS or more generally to robotics.

Reviewer 3: All reporting to happen through the security@ email address; the security WG's best communication vehicle is our matrix chat room. The person/people triaging incoming email to that address should also be included in the Security WG matrix chat room, and simply call to the chat room for any support needed.

So if the triage person is not able to quickly handle the report, they should call out for help in matrix.

> The ROS Security Working Group may also be able to assist you with reporting vulnerabilities which are not in scope of this policy but which are related to ROS or more generally to robotics.

Reviewer: > All reporting to happen through the security@ email address

This part of the REP is discussing reporting things that are _not_ in scope. In that sense, do we really want them still reporting to security@? I'd say the reporter should contact the WG on their own (in matrix or via our [google group](https://groups.google.com/forum/#!forum/ros-security)) for help on issues outside the scope of the VDP, or security@ could get swamped with out of scope issues.

> The ROS Security Working Group may also be able to assist you with reporting vulnerabilities which are not in scope of this policy but which are related to ROS or more generally to robotics.

Reviewer 3: Right now this doc only gives the reader a single entry point which is the security@ email. If we want reports elsewhere, we need to add another entry point. The google group is private, and I think we want to keep the matrix room private?

Agreed the security@ email could get spammed, but it should only be with security related issues. I would imagine OR would manage this similar to abuse@ and postmater@ email aliases.

> The ROS Security Working Group may also be able to assist you with reporting vulnerabilities which are not in scope of this policy but which are related to ROS or more generally to robotics.

Reviewer: > If we want reports elsewhere, we need to add another entry point. The google group is private, and I think we want to keep the matrix room private?

Neither are private, nor should they be. That's one of the reasons we don't want the WG to be driving vuln handling-- we can't handle embargo in a public group, and we want the security WG to be public.

> The ROS Security Working Group may also be able to assist you with reporting vulnerabilities which are not in scope of this policy but which are related to ROS or more generally to robotics.

Reviewer 3: The Security WG has always seemed very open to discussing security issues, but looking at it now I don't see a public facing, easily accessible "contact us" landing place. I've added an agenda item to discuss public interaction at the next WG meeting and would propose leaving this as-is for the time being. Once the WG settles on an approach for public interaction, we can open a PR on how to interact with the Security WG.

> The ROS Security Working Group may also be able to assist you with reporting vulnerabilities which are not in scope of this policy but which are related to ROS or more generally to robotics.

Author: We've changes this even more down below.  @SidFaber @kyrofa is there more to be done for this particular item, or can we resolve this conversation?

An increasing number of companies are bringing ROS-based products to market, and during this growth period we anticipate vulnerabilities to be identified in products before vendors have an established vulnerability reporting program.
We will work with you on a best-effort basis to help connect you with the responsible vendor.

How to submit a vulnerability
=============================

Please submit vulnerability reports by emailing security@openrobotics.org, preferably in English.
This is a private list of security officers who can help verify the bug report and develop and release a fix.
The more information provided about the bug, the easier it will be to fix.
If you already have a fix, please include it with your report.
We may ask to exchange PGP keys to discuss sensitive details about the vulnerability which may include proof-of-concept code, an impact assessment, recommended remediation steps or other information that will help us find and fix the problem.

Include any plans you may have to disclose details about the vulnerability.

> In order to protect our users, we ask that you not publicly discuss the vulnerability until at least 90 days have passed or the fix is published.

Author: Yeah, good point.  Will update that.

We will notify you when a pull request has been submitted and approved which remediates the vulnerability.
Because security vulnerabilities can have severe consequences, they differ from other bugs and at our discretion-- in order to protect our users-- we may modify our normal code release processes for publishing open source code updates.

Safe Harbor
===========

Open Robotics strongly supports security research into ROS software and seeks to encourage that research.
Open Robotics will not engage in legal action against individuals who act in good faith to identify, report and fix vulnerabilities in ROS, so long as they operate in accordance with any applicable laws as well as this policy.
Unauthorized research or testing against operating robotic systems is in violation of this policy and strongly discouraged due to potential health and human safety concerns.

If at any time you have concerns about whether your activities are consistent with this policy, please contact us at security@openrobotics.org.

Copyright
=========

> This document is placed in the public domain or under the CC0-1.0-Universal license, whichever is more permissive.

Author: I copied that from REP-2004: https://github.com/ros-infrastructure/rep/blame/master/rep-2004.rst#L852 .  My understanding is that there is no public domain in some jurisdictions in the world, so the "or" clause handles those.  But I really don't feel strongly about it either way.

> This document is placed in the public domain or under the CC0-1.0-Universal license, whichever is more permissive.

Reviewer 2: I wouldn't know the legal implications of one or the other. As far as I can tell *EVERY REP but REP-2004* uses "This document has been placed in the public domain." so it is surprising to see new verbiage appear. If the new verbiage is needed, shouldn't the other REPs be updated accordingly ?

I don't have an opinion about which one is used but as we were several bringing up the inconsistency in the Security Working group yesterday it seemed relevant to point it out here.

> This document is placed in the public domain or under the CC0-1.0-Universal license, whichever is more permissive.

Reviewer: Yeah I don't think anyone has strong feelings either way other than feeling strongly about consistency.

> This document is placed in the public domain or under the CC0-1.0-Universal license, whichever is more permissive.

Author: I've switched the verbiage just to the "public domain" for now.  We can revisit it as a batch later if needed.

> This document is placed in the public domain or under the CC0-1.0-Universal license, whichever is more permissive.

Reviewer 4: This also came up during the review of `2004`. See the last comment [here](https://github.com/ros-infrastructure/rep/pull/218/files/7640cce93f5c0b140faa8a919beb1a2870f52896#diff-f40b01badf3f268cbb5d72ffcc9613f5) (I can't seem to link directly to *resolved* discussions).

> This document is placed in the public domain or under the CC0-1.0-Universal license, whichever is more permissive.

Author: Sigh.  Given the rationale in that PR, I'm going to revert 7626fe9 .  Note that I am ignoring further comments on this particular topic; it's not worth the time.

Status: Active
Type: Informational
Content-Type: text/x-rst
Created: 12-May-2020
Post-History:

Abstract
========

This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.

Motivation
==========

The developers of the Robot Operating System (ROS) take security very seriously.
As such, we would like to be informed when a security bug is found so that it can be fixed and disclosed as quickly as possible.
The rest of the document outlines what is covered by this policy and how to report security vulnerabilities.

Scope
=====

The Vulnerability Disclosure Program outlined here covers all code within the ROS 2 Common Packages as defined in REP 2005.

The ROS Security Working Group may also be able to assist you with reporting vulnerabilities which are not in scope of this policy but which are related to ROS or more generally to robotics.
An increasing number of companies are bringing ROS-based products to market, and during this growth period we anticipate vulnerabilities to be identified in products before vendors have an established vulnerability reporting program.

> We will work with you on a best-effort basis to help connect you with the responsible vendor.

Reviewer 3: Understood--the reporter simply sees a vulnerability, but that vuln could have been introduced anywhere along the supply chain. Although the scenario is complex, this paragraph interpreted as "best effort" only, not a commitment to any specific actions or timeline, and the burden should fall mainly on individuals receiving inbound reports.

Although the policy could stand without this paragraph, it's included specifically for outreach. It seems we should help address the current dearth of robot vuln reporting (of which you're well aware!). If we're able to kickstart reporting to the point where this becomes burdensome, we can claim success and update the REP to provide hands-off redirection. To my knowledge the most relevant and active CERT would be [ICS-CERT](https://www.us-cert.gov/ics) although they seem primarily focused on embedded systems.

Good point on the suggested change--the vendor may not be the one to address. Does this help clarify further?
```suggestion
We will work with you on a best-effort basis to help connect you with responsible parties best suited to address your concerns.
```

> We will work with you on a best-effort basis to help connect you with the responsible vendor.

Author: > Who's going to be in charge of maintaining a list of of affected vendors (per ROS software package I guess) and distributing the information coherently? Open Robotics? If so, this will require a considerable amount of resources.

No, Open Robotics is definitely not going to commit to this.

This section has seemed to cause a lot of confusion up until this point.  To me, it is fairly clear that this is a "we might be able to help you find the responsible parties, since we know the robotics community.  But we don't guarantee anything".  But others aren't reading it this way.  I guess we can go with @vmayoral's suggestion here, but I'm thinking we should just remove this paragraph completely.  @ros2/security_working_group Thoughts?

> We will work with you on a best-effort basis to help connect you with the responsible vendor.

Reviewer 5: Having the chance to derive the load to third parties is the way to go. Either @SidFaber's suggestion or the one I initially propose would do. ICS-CERT is definitely the most popular but we're pretty aligned with [INCIBE-CERT](https://www.incibe-cert.es/en) and have interacted with a few others in EU. There're [big lists of reponse teams out there](https://www.first.org/members/teams/#CERT) and we probably should make a list of which ones should be considered (though this is a Security WG conversation).

How to submit a vulnerability
=============================

Please submit vulnerability reports by emailing security@openrobotics.org, preferably in English.
This is a private list of security officers who can help verify the bug report and develop and release a fix.
The more information provided about the bug, the easier it will be to fix.
If you already have a fix, please include it with your report.
We may ask to exchange PGP keys to discuss sensitive details about the vulnerability which may include proof-of-concept code, an impact assessment, recommended remediation steps or other information that will help us find and fix the problem.

Include any plans you may have to disclose details about the vulnerability.

> In order to protect our users, we ask that you not publicly discuss the vulnerability until at least 90 days have passed or the fix is published.

Author: Works for me, I'll incorporate this suggestion.

The group handling the report will also follow this timeline.
If you do not wish to be acknowledged in the release communications please indicate so when you submit the vulnerability.

What to expect in response to a vulnerability disclosure
========================================================

Expect a timely response to your notification, normally within two business days, during which time we will triage your vulnerability report.

> After we have evaluated the vulnerability we will send you our risk assessment and, where applicable, the anticipated time to publish an update.

Author: > The best scenario is one where Open Robotics becomes an accreddited CNA for ROS-related flaws, the same as Alias Robotics or Canonical, and directly books a CVE ID block, assigning as appropriate.

This isn't something Open Robotics can commit to at this time.

While CVE is indeed the current system that many people use, I've often heard stirrings of something different coming along.  Thus, I'd prefer to be a little more generic here with something like:

```
Where appropriate, we will coordinate and assign a vulnerability ID with an industry-standard vulnerability database such as CVE.
```

@ros2/security_working_group thoughts?

> After we have evaluated the vulnerability we will send you our risk assessment and, where applicable, the anticipated time to publish an update.

Reviewer 3: CVEs have come up often enough that it's probably a good idea to explicitly address them and this seems like a good solution.

> After we have evaluated the vulnerability we will send you our risk assessment and, where applicable, the anticipated time to publish an update.

Reviewer 5: Agreed with @SidFaber, IMHO we should tackle this here. From a security angle and as far as I know, this is the best first step an organization/group can go to commit on security for their products. 

@clalancette I'd insist it's best for Open Robotics to get acredited as a CNA, get the right training, define the scope (possibly including ROS, ROS 2 and Gazebo) while committing some person-power to this. I don't mind walking you through this myself since we did it last year. 

If this is still not an option, Alias can commit to contribute with some resources during the following first months (we'd be reporting flaws ourselves so ... it kind of like makes sense) but we'd need to eventually either share the workload with someone or find resources to continue doing it.

> After we have evaluated the vulnerability we will send you our risk assessment and, where applicable, the anticipated time to publish an update.

Author: > @clalancette I'd insist it's best for Open Robotics to get acredited as a CNA, get the right training, define the scope (possibly including ROS, ROS 2 and Gazebo) while committing some person-power to this. I don't mind walking you through this myself since we did it last year.

I will discuss this internally, but we should proceed for now as if Open Robotics isn't going to do this.  Security isn't our main focus or our expertise.  This is part of the reason for having a security working group; so we can have security experts who know how to deal with this.

Regardless, this isn't a blocking issue for this Vulnerability Disclosure.  This gets more into the "internal processes" stuff that @SidFaber is putting together.

> After we have evaluated the vulnerability we will send you our risk assessment and, where applicable, the anticipated time to publish an update.

Author: All right, I've added slightly more generic language to this section in 7371868.  I'm going to resolve this conversation for now, but feel free to reopen or keep commenting.

We will aim at being as transparent as possible about timelines and any issues or challenges which may impact the scheduled fix.

Should the vulnerability have a widespread impact that requires coordinated disclosure, we may, at our discretion, choose to engage a neutral third party such as CERT/CC or ICS-CERT to support coordination efforts.

We will notify you when a pull request has been submitted and approved which remediates the vulnerability.
Because security vulnerabilities can have severe consequences, they differ from other bugs and at our discretion-- in order to protect our users-- we may modify our normal code release processes for publishing open source code updates.

Safe Harbor
===========

Open Robotics strongly supports security research into ROS software and seeks to encourage that research.

> Open Robotics will not engage in legal action against individuals who act in good faith to identify, report and fix vulnerabilities in ROS, so long as they operate in accordance with any applicable laws as well as this policy.

Reviewer 5: https://www.first.org/global/sigs/vulnerability-coordination/multiparty/guidelines-v1.1

> Open Robotics will not engage in legal action against individuals who act in good faith to identify, report and fix vulnerabilities in ROS, so long as they operate in accordance with any applicable laws as well as this policy.

Author: The suggestion seems reasonable to me.  @ros2/security_working_group, what do you think?

> Open Robotics will not engage in legal action against individuals who act in good faith to identify, report and fix vulnerabilities in ROS, so long as they operate in accordance with any applicable laws as well as this policy.

Reviewer 3: @vmayoral, the change above seems OK to me if OR is good with it--do you propose any other changes to the doc or is this sufficient?

> Open Robotics will not engage in legal action against individuals who act in good faith to identify, report and fix vulnerabilities in ROS, so long as they operate in accordance with any applicable laws as well as this policy.

Reviewer 5: I think that'll do.

REP: 2006
Title: ROS 2 Vulnerability Disclosure Policy
Author: ROS 2 Security Working Group, Chris Lalancette <clalancette@openrobotics.org>
Status: Active
Type: Informational
Content-Type: text/x-rst
Created: 12-May-2020
Post-History:

Abstract
========

> This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.

Author: I swear I had done this, but either I lost it in my editor or I forgot.  Anyway, good call, I'll add that now.

> This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.

Author: 52ddc96

The rest of the document outlines what is covered by this policy and how to report security vulnerabilities.

Scope
=====

The Vulnerability Disclosure Program outlined here covers all code within the ROS 2 Common Packages as defined in REP 2005.

The ROS Security Working Group may also be able to assist you with reporting vulnerabilities which are not in scope of this policy but which are related to ROS or more generally to robotics.
An increasing number of companies are bringing ROS-based products to market, and during this growth period we anticipate vulnerabilities to be identified in products before vendors have an established vulnerability reporting program.
We will work with you on a best-effort basis to help connect you with responsible parties best suited to address your concerns.

How to submit a vulnerability
=============================

Please submit vulnerability reports by emailing security@openrobotics.org, preferably in English.
This is a private list of security officers who can help verify the bug report and develop and release a fix.
The more information provided about the bug, the easier it will be to fix.

> If you already have a fix, please include it with your report.

Reviewer 6: Should we specify a favored format for the fix (e.g. git patch)? Similarly to the favored language for the mail.

Reviewer 7: +1 I just mentioned this in my comment as well

Author: My opinion is that adding too much structure at the beginning can scare people off, and isn't really necessary.  I'm going to (perhaps naively) assume that security researches know how to send a patch.  If we start consistently getting h.264-encoded patches or something, we can amend this to add more structure.

Reviewer 7: Is this only designed for established pentesters? What if some junior-dev suddenly discovers a behavior of its ROS2 robot that he considers security relevant and wants to escalate to the security WG? I would suggest not adding a strong typed format but maybe a suggestion for "newbies". Therefor, it would be easier to flag false positives and kindly point out a wrong config or so.

Author: I don't really see the situation as being much different for junior people.  If they don't know how to send a patch, then we'll get something that is maybe not useful.  But my opinion doesn't change here; unless/until this becomes a common occurrence, I don't really see a need.  Maybe people in the security working group disagree.

Reviewer 6: As per [my comment earlier](https://github.com/ros-infrastructure/rep/pull/262#issuecomment-632755837) this is something we are going to discuss during the next WG meeting - not the patch format but if we should set a report template at all.

We may ask to exchange PGP keys to discuss sensitive details about the vulnerability which may include proof-of-concept code, an impact assessment, recommended remediation steps or other information that will help us find and fix the problem.

Include any plans you may have to disclose details about the vulnerability.

> In order to protect our users, unless otherwise agreed by both parties, we ask that you not publicly discuss the vulnerability until at least 90 days have passed or the fix is published.

Reviewer 6: > 90 days have passed

since? The first mail I assume? I think it should be explicit anyways.

Author: Yeah, good point.  I'll make that explicit.

> In order to protect our users, unless otherwise agreed by both parties, we ask that you not publicly discuss the vulnerability until at least 90 days have passed from the initial contact with us or the fix is published.

Reviewer 8: @clalancette nit: 
```suggestion
In order to protect our users, unless otherwise agreed by both parties, we ask that you do not publicly discuss the vulnerability until at least 90 days have passed from the initial contact with us or the fix is published.
```

Reviewer 8: I'd also suggest:
```suggestion
In order to protect our users, unless otherwise agreed by both parties, we ask that you not publicly discuss the vulnerability until a fix is published or at least 90 days have passed since the initial report submission.
```
It find it clearer, but I'm not a native speaker so feel free to disagree/disregard.

Author: I took your second suggestion, since I agree it is clearer.  The first one I left out, because I actually think it is clearer *without* it :).

The group handling the report will also follow this timeline.
If you do not wish to be acknowledged in the release communications please indicate so when you submit the vulnerability.

What to expect in response to a vulnerability disclosure
========================================================

Expect a timely response to your notification, normally within two business days, during which time we will triage your vulnerability report.

After we have evaluated the vulnerability we will send you our risk assessment and, where applicable, the anticipated time to publish an update.

> Where appropriate, we will coordinate and assign a vulnerability ID with an industry-standard vulnerability database such as CVE.

Reviewer 5: Just a minor aspect but to make sure we don't get yelled at by other security groups, we may want to consider  the following given that CVE is not a vulnerability database. As per the official definition, 

> Common Vulnerabilities and Exposures (CVE) List CVE® is an archive (dictionary according to the official source) of entries—each containing an identification number, a description, and at least one public reference—for publicly known cybersecurity vulnerabilities. 

You might be referring to NVD:

> U.S. National Vulnerability Database (NVD) is the U.S. government repository of standards based vulnerability management data. It presents an archive with vulnerabilities, each with their corresponding CVE identifiers. NVD gets fed by the CVE List and then builds upon the information included in CVE Entries to provide enhanced information for each entry such as fix information, severity scores, and impact ratings.

To avoid verbosity, I'd suggest to keep it simple :)
```suggestion
Where appropriate, we will coordinate and assign a vulnerability ID with an industry-standard.
```

We will aim at being as transparent as possible about timelines and any issues or challenges which may impact the scheduled fix.

Should the vulnerability have a widespread impact that requires coordinated disclosure, we may, at our discretion, choose to engage a neutral third party such as CERT/CC or ICS-CERT to support coordination efforts.

We will notify you when a pull request has been submitted and approved which remediates the vulnerability.
Because security vulnerabilities can have severe consequences, they differ from other bugs and at our discretion-- in order to protect our users-- we may modify our normal code release processes for publishing open source code updates.

Safe Harbor
===========

Open Robotics strongly supports security research into ROS software and seeks to encourage that research.
Open Robotics will not engage in legal action against individuals who act in good faith to identify, report and fix vulnerabilities in ROS, so long as they operate in accordance with any applicable laws or this policy.

> Unauthorized research or testing against operating robotic systems is in violation of this policy and strongly discouraged due to potential health and human safety concerns.

Reviewer 9: The term of `Unauthorized` doesn't seem to fit. This makes it sound like prior clearance is required.
E.g. How would a third party gain proper authorization to research or test ROS?

Author: I took this to mean "Unauthorized by the owner/operator of the robot", but I'll add a little something to clarify this.

Author: 52ddc96
